lib/compress/zstd_ldm.c

```diff
@@ -299,10 +295,13 @@ static size_t ZSTD_ldm_generateSequences_internal(
     U64 rollingHash = 0;
 
     while (ip <= ilimit) {
+        U32 const currentOffset = (U32)(ip - base);
```

Comment:
nit: this is not an offset it is an "index". Either keep `curr` or use `currentIndex` (note: `current` cannot be used because it conflicts with a macro in the Linux kernel).

I know that it is called `offset` in the `ldmEntry_t` struct, but that is a bad name. It should be renamed to `index`. If you want to include that renaming in this diff or a followup that would be great.